src/core/border_router/routing_manager.cpp

```diff
@@ -221,14 +240,70 @@ Error RoutingManager::LoadOrGenerateRandomBrUlaPrefix(void)
 }
 
 #if OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
+const otIp4Address RoutingManager::kWellKnownIpv4OnlyAddress1 = {{{192, 0, 0, 170}}};
+const otIp4Address RoutingManager::kWellKnownIpv4OnlyAddress2 = {{{192, 0, 0, 171}}};
+
+void RoutingManager::DiscoverInfraIfNat64Prefix(void)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(IsInitialized() && mInfraIf.IsRunning(), error = kErrorInvalidState);
+
+    LogInfo("Discovering infrastructure NAT64 prefix on %s", mInfraIf.ToString().AsCString());
+    error = mInfraIf.DiscoverNat64Prefixes();
+
+exit:
+    if (error != kErrorNone)
+    {
+        LogWarn("Failed to request infrastructure NAT64 prefix on %s: %s", mInfraIf.ToString().AsCString(),
+                ErrorToString(error));
+    }
+}
+
+void RoutingManager::DiscoverInfraIfNat64PrefixesDone(const Ip6::Address *aAddresses, uint8_t aNumAddresses)
+{
+    Ip6::Prefix   prefix;
+    const uint8_t prefixLength[] = {32, 40, 48, 56, 64, 96};
```

Comment:
Define as `static` and also use `kPrefixLength` (since it is constant)?

```suggestion
    static const uint8_t kPrefixLength[] = {32, 40, 48, 56, 64, 96};
```

Reply:
Updated